src/gui/qgsmapcanvas.cpp

```diff
@@ -3271,7 +3271,7 @@ void QgsMapCanvas::readProject( const QDomDocument &doc )
       // never manually set the crs for the main canvas - this is instead connected to the project CRS
       setDestinationCrs( tmpSettings.destinationCrs() );
     }
-    setExtent( tmpSettings.extent() );
```

Comment:
Why are you removing this?

What we have to do here:
If the option is activated, we have to transform the extent to the CRS of the canvas and use it.
If the options is not activated, keep the existing code.